packages/api-plugin-promotions/src/resolvers/Mutation/archivePromotion.js

```diff
@@ -10,7 +10,8 @@
  */
 export default async function archivePromotion(_, { input }, context) {
   const { promotionId, shopId } = input;
-  await context.validatePermissions("reaction:legacy:promotions", "update", { shopId });
+  await context.validatePermissions("reaction:legacy:promotions", "archive", { shopId });
```

Comment:
What is the reasoning behind wanting to have a separate permission besides "update"?

Reply:
That's the requirements from this ticket https://app.zenhub.com/workspaces/moc-main-board-62fd8e881686130027eafc31/issues/reactioncommerce/kinetic/111. Brendan said we need a separate permission for archived promotions

Reply:
I see statements regarding permissions to view archived but nothing about the ability to archive?

Reply:
That being said I think it makes sense to add the additional permission, wondering if we should add "unarchive" functionality

Reply:
If we add permissions these need to be added to the default roles in this PR

Reply:
https://github.com/reactioncommerce/reaction/blob/5eb585913b0c0ec79682355849fb6168c537ba1c/packages/api-plugin-authorization-simple/src/util/defaultRoles.js#L21

Reply:
Yeah, I thought we should have separate permissions for archive(view/update), but seems like we only need additional permission for read archived promotions. I will update the PR. Thanks

Reply:
@vannguyenn If you read my comments I do agree we should have a separate permission for "archive". Do you disagree?

Reply:
Oh, I thought you said that we only need the additional view archived permission based on the statements in the ticket. I will revert my changes. Thanks